### PR TITLE
IOCTL support in python-fuse

### PR DESCRIPTION
**Pull request by beg0**
Hi.
I wrote a patch to add support of ioctl in the libfuse python binding.
I also wrote an example file (fioc.py) that does the same things as example/fioc.c does in libfuse's examples code. It works with the counterpart example/fioclient.c in libfuse examples code.

To wrote an example code that do not depend on non-standard python module such as ioctl-opt, I rewrote a class to compile IOCTL values. Unfortunately this code may have portability issues.
Let me know if you have any comments
Cédric

### CODE REVIEW

**Comment by sdelafond** [comment]
Hello, this is not mergeable as-is, since python-fuse was recently modified to be compatible with python 3. Would you have any interest in adapting your PR to bring IOCTL support against the most recent version ?

**Comment by beg0** [comment]
I will have a look at the new code for python-fuse and try to merge my modifications.

**Comment by beg0** [comment]
Hi. I just finished updating the IOCTL related code to work with python3. Please review the code and tell me if everything is fine for you.
Best Regards

File: example/fioc.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    def do_write(self, path, buf, size, offset):
+        self.buf = self.buf[0:offset-1] + buf + self.buf[offset+size+1:len(self.buf)]
+
+    def write(self, path, buf, size, offset):
+        if self.file_type(path) != FIOC_FILE:
+            return -errno.EINVAL;
+
+        self.do_write(path, buf, size, offset)
+
+    def trunate(self, path, size):
```
Comment (sdelafond) [inline example/fioc.py]: s/trunate/truncate/ I guess ?

Comment (beg0) [reply example/fioc.py]: yes, it's a typo.

Comment (sdelafond) [reply example/fioc.py]: Can you correct it and update the PR ?

**Comment by sdelafond** [comment]
Thanks for your contribution.

**Pull request merged**